test/acceptance/features/environment.py

```diff
@@ -55,3 +62,26 @@ def before_scenario(_context, _scenario):
     _context.bindings = dict()
     output, code = cmd.run(f'{ctx.cli} get ns default -o jsonpath="{{.metadata.name}}"')
     assert code == 0, f"Checking connection to OS cluster by getting the 'default' project failed: {output}"
+
+
+def after_scenario(context, scenario):
+    global scenario_status
+    context.scenario_status = scenario_status
+    namespace = os.getenv("TEST_NAMESPACE")
+    if delete_namespace == "alwaysdelete":
+        if scenario.status == Status.passed:
```

Comment:
If you want to "always delete", it does not make sense to check for scenario status... in this configuration the namespace must be deleted no matter what (even when scenario failed).